sources/packages/web/src/views/aest/institution/Programs.vue

```diff
@@ -1,13 +1,213 @@
+          </template>
+        </Column>
+        <Column field="programStatus" header="Status"
+          ><template #body="slotProps">
+            <Chip
+              :label="slotProps.data.programStatus"
+              class="p-mr-2 p-mb-2 p-text-uppercase"
+              :class="
+                getProgramStatusColorClass(slotProps.data.programStatus)
+              "/></template
+        ></Column>
+        <Column>
+          <template #body="slotProps">
+            <v-btn
+              outlined
+              @click="goToViewProgramDetail(slotProps.data.programId)"
+              >View</v-btn
+            >
+          </template>
+        </Column>
+      </DataTable>
+    </content-group>
   </content-group>
 </template>
 
 <script lang="ts">
+import { onMounted, ref, computed } from "vue";
+import { useRouter } from "vue-router";
+import { InstitutionService } from "@/services/InstitutionService";
 import ContentGroup from "@/components/generic/ContentGroup.vue";
+import {
+  AESTInstitutionProgramsSummaryPaginatedDto,
+  ApprovalStatus,
+} from "@/types";
+import { useToastMessage } from "@/composables";
+import { AESTRoutesConst } from "@/constants/routes/RouteConstants";
 export default {
   components: { ContentGroup },
-  //TODO, create and fetch an API to fetch Programs
+  props: {
+    institutionId: {
+      type: Number,
+      required: true,
+    },
+  },
+  setup(props: any) {
+    const toast = useToastMessage();
+    const router = useRouter();
+    const institutionProgramsSummary = ref(
+      {} as AESTInstitutionProgramsSummaryPaginatedDto,
+    );
+    const searchProgramName = ref("");
+    const defaultPage = 0;
+    const defaultNoOfRows = ref(10);
+    const defaultSortOrder = -1;
+    const loading = ref(false);
+    onMounted(async () => {
+      institutionProgramsSummary.value = await InstitutionService.shared.getPaginatedAESTInstitutionProgramsSummary(
+        props.institutionId,
+        defaultNoOfRows.value,
+        defaultPage,
+        defaultSortOrder,
+        searchProgramName.value,
+      );
+      if (institutionProgramsSummary.value.programsSummary.length === 0) {
+        toast.warn(
+          "No Programs found",
+          "No Programs found for the Institution",
+        );
+      }
+    });
+    const programsFound = computed(() => {
+      return institutionProgramsSummary.value.programsCount > 0;
+    });
+    const goToViewProgramDetail = (programId: number) => {
+      router.push({
+        name: AESTRoutesConst.PROGRAM_DETAILS,
+        params: { programId: programId },
+      });
+    };
+    const getProgramStatusColorClass = (status: ApprovalStatus) => {
+      switch (status) {
+        case ApprovalStatus.approved:
+          return "bg-info text-white";
+        case ApprovalStatus.pending:
+          return "bg-warning text-white";
+        default:
+          return "";
+      }
+    };
+
+    const onPage = async (event: any) => {
+      loading.value = true;
```

Comment:
In general, methods that are dealing specifically with the API call preparation or the logic around it could be added to the service layer to leave the UI cleaner.